### ISSUE TITLE
Slopes in maps

### ISSUE THREAD

**Issue by codeandkey** [codeandkey/boom#115]
It's about time we added stairs. It may require a refactor of the collision system but it should be feasible to have Terraria-esque autoclimbing in the game.

**Comment by quigley-c** [comment]
How were you imagining this @codeandkey? Would it benefit us more to use half-height platforms or try to implement slanted collision?

**Comment by codeandkey** [comment]
I think we should keep the player decoupled from the physics engine and roll our own collisions. A 1/8 or 1/4 size block might be a good size for the collision resolution, which should allow us to create stairs and other slopes.

**Comment by codeandkey** [comment]
With much optimization it also might be worth it to just implement pixel-perfect collisions on a certain world layer.
